### PR TITLE
Add `@web-stories-wp` prefix to karma packages name

### PR DESCRIPTION
**Pull request by mohdsayed**
## Context

Add `@web-stories-wp` prefix to karma packages name.

### QA

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Related: https://github.com/google/web-stories-wp/pull/8537#issuecomment-891263013

### CODE REVIEW

**Comment by swissspidy** [comment]
Now we can add them to `package.json` as well

File: karma-dashboard.config.cjs [hunk 1]
```diff
@@ -57,7 +57,7 @@ module.exports = function (config) {
 
     // Frameworks to use.
     // Available frameworks: https://npmjs.org/browse/keyword/karma-adapter
-    frameworks: ['jasmine', 'karma-puppeteer-client'],
+    frameworks: ['jasmine', '@web-stories-wp/karma-puppeteer-client'],
```
Comment (swissspidy) [inline karma-dashboard.config.cjs:60]: There's currently an error:

```
Server start failed on port 9876: Error: No provider for "framework:@web-stories-wp/karma-puppeteer-client"! (Resolving: framework:@web-stories-wp/karma-puppeteer-client)
```

We'll need to update `plugins` above too:

```diff
-require('./packages/karma-puppeteer-launcher/src/index.cjs'),
-require('./packages/karma-puppeteer-client/src/index.cjs'),
-require('./packages/karma-cuj-reporter/src/index.cjs'),
-require('./packages/karma-failed-tests-reporter/src/index.cjs'),
+require('@web-stories-wp/karma-puppeteer-launcher'),
+require('@web-stories-wp/karma-puppeteer-client'),
+require('@web-stories-wp/karma-cuj-reporter'),
+require('@web-stories-wp/karma-failed-tests-reporter'),
```

And of course add the packages to `package.json`. Then it should work.

Comment (mohdsayed) [reply karma-dashboard.config.cjs:60]: It still throws the same error

```bash
03 08 2021 14:27:21.283:ERROR [karma-server]: Server start failed on port 9877: Error: No provider for "framework:@web-stories-wp/karma-puppeteer-client"! (Resolving: framework:@web-stories-wp/karma-puppeteer-client)
03 08 2021 14:27:21.283:ERROR [karma-server]: Server start failed on port 9876: Error: No provider for "framework:@web-stories-wp/karma-puppeteer-client"! (Resolving: framework:@web-stories-wp/karma-puppeteer-client)
ERROR: "test:karma:edit-story" exited with 1.
```

File: package.json [hunk 2]
```diff
@@ -195,7 +195,11 @@
     "webpack": "^4.46.0",
     "webpack-bundle-analyzer": "^4.4.2",
     "webpack-cli": "^4.7.0",
-    "webpackbar": "^4.0.0"
+    "webpackbar": "^4.0.0",
+    "@web-stories-wp/karma-cuj-reporter": "*",
```
Comment (swissspidy) [inline package.json]: Alphabetical order -> these should be on line ~121

File: karma-dashboard.config.cjs (hunk 1 above)
Comment (mohdsayed) [reply karma-dashboard.config.cjs:60]: This was missing https://github.com/google/web-stories-wp/pull/8558/commits/cbe3846496174d47cdfbf8ff3bd584ae3a1b5720

Comment (swissspidy) [reply karma-dashboard.config.cjs:60]: Hmm bummer.

According to the [docs](http://karma-runner.github.io/6.3/config/plugins.html) loading the plugins like that should work. It [even supports weildcards](https://github.com/karma-runner/karma/pull/3659) for that.

But might be missing something here 😕 Perhaps we have to dig into the plugin resolution a little bit to see where it's going wrong.

`node_modules/karma/lib/plugin.js` seems to be the file responsible for the loading of plugins.

Comment (mohdsayed) [reply karma-dashboard.config.cjs:60]: But exporting module key with the scope name solves the problem https://github.com/google/web-stories-wp/commit/cbe3846496174d47cdfbf8ff3bd584ae3a1b5720

Comment (swissspidy) [reply karma-dashboard.config.cjs:60]: Woohoo!

Does this need to be done for the other packages' exports as well?

Comment (mohdsayed) [reply karma-dashboard.config.cjs:60]: I cannot find any documentation for that here https://karma-runner.github.io/5.2/dev/plugins.html
But what I can tell by looking at the examples of plugins is that the name of the export module key should be the same as being used in reporters/framework/launchers array.

https://github.com/karma-runner/karma-growl-reporter/blob/master/index.js#L69
https://github.com/karma-runner/karma-growl-reporter/blob/master/README.md

```js
module.exports = function(config){
    config.set({
        reporters: ['progress','growl'],
   });
};
```

Currently we are using key names like 
```
'reporter:cuj'
'reporter:failed-tests'
'launcher:puppeteer'
```

I think we should keep it consistent with the package names.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**